src/Shared/UnitTests/ObjectModelHelpers.cs

```diff
@@ -243,7 +243,10 @@ public static void AssertItems(string[] expectedItems, IList<TestItem> items, Di
                 expectedItems.ShouldNotBeEmpty();
             }
 
-            for (var i = 0; i < expectedItems.Length; i++)
+            // iterate to the minimum length; if the lengths don't match but there's a prefix match the count assertion below will trigger
```

Comment:
Could we just move the assert up? Seems simpler.

Reply:
That was my initial implementation, but it loses a bunch of information in the error message--it's nicer to see the item-level diff if there is one. That helped me chase down a couple of errors in this implementation.